prov/rxm/src/rxm_cq.c

```diff
@@ -607,39 +607,53 @@ void rxm_cq_write_error(struct util_cq *cq, struct util_cntr *cntr,
 		assert(0);
 }
 
-static size_t rxm_cq_read_write_error(struct fid_cq *msg_cq,
-				      struct fi_cq_data_entry *comp,
-				      ssize_t err)
+static void rxm_cq_write_error_all(struct rxm_ep *rxm_ep, int err)
+{
+	struct fi_cq_err_entry err_entry = {0};
+	ssize_t ret = 0;
+
+	err_entry.prov_errno = err;
+	err_entry.err = err;
+	if (rxm_ep->util_ep.tx_cq) {
+		ret = ofi_cq_write_error(rxm_ep->util_ep.tx_cq, &err_entry);
+		if (ret)
+			assert(0);
+	}
+	if (rxm_ep->util_ep.rx_cq) {
+		ret = ofi_cq_write_error(rxm_ep->util_ep.rx_cq, &err_entry);
+		if (ret)
+			assert(0);
+	}
+	if (rxm_ep->util_ep.flags & OFI_CNTR_ENABLED) {
+		rxm_cntr_incerr(rxm_ep->util_ep.tx_cntr);
+		rxm_cntr_incerr(rxm_ep->util_ep.rx_cntr);
+		rxm_cntr_incerr(rxm_ep->util_ep.wr_cntr);
+		rxm_cntr_incerr(rxm_ep->util_ep.rd_cntr);
```

Comment:
This isn't necessary true, though.  Not all counters should increment.

Reply:
Yes but we don't have the context to determine which counter to increment. Also, I should check if a counter is valid before incrementing it. I'll add it.

Reply:
Yeah, I figured out what was happening after writing the comment.